lib/active_model/validations/store_model_validator.rb

```diff
@@ -24,7 +24,11 @@ def validate_each(record, attribute, value)
         when :json
           strategy.call(attribute, record.errors, value.errors) if value.invalid?
         when :array
-          record.errors.add(attribute, :invalid) if value.any?(&:invalid?)
+          value.each_with_index do |element, index|
+            if element.invalid?
+              strategy.call(attribute, record.errors, element.errors, "[#{index}] ")
+            end
+          end
```

Comment:
Here is the fix. Along the way we're getting merge_errors support for arrays too